halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/ValidationRun.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.halyard.deploy.services.v1;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.*;
+import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import com.netflix.spinnaker.halyard.core.problem.v1.Problem;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.ApplicationContext;
+
+@RequiredArgsConstructor
+public class ValidationRun {
```

Comment:
This class is hard to understand, it looks like it started thinking in a `Runnable` but it's not.
Most of this logic can be moved to the `ValidationService` instead, and `ValidatorCollection` also has methods for running all validators against a `Node` tree, so it could be enhanced for this use case instead of creating different logic. For example, you wouldn't need to change or use `NodeFilter`.

Reply:
It started this way but then you have to pass parameters everywhere which is not ideal. I feel this is cleaner to instantiate a validation. `ValidationCollection` iterates through nodes and validators differently (goes through each Node's class hierarchy).